### ISSUE TITLE
Onnx CLIP Model outputs "Shape mismatch" warining on inference

### ISSUE THREAD

**Issue by abnokubi** [huggingface/transformers#19937]
### System Info

- `transformers` version: 4.23.1
- Platform: Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.29
- Python version: 3.8.10
- Huggingface_hub version: 0.10.1
- PyTorch version (GPU?): 1.12.1+cu116 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@patil-suraj

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. Convert clip to onnx `python -m transformers.onnx -m openai/clip-vit-base-patch32 onnx/`
2. Try to inference

```
import transformers
import onnxruntime
import numpy as np
from PIL import Image
import torch
model_path = "onnx/model.onnx"
example_image = "swatch01.png"
session = onnxruntime.InferenceSession(model_path)
processor = transformers.CLIPProcessor.from_pretrained("openai/clip-vit-base-patch32")
labels = [
            "a close up of a person's eye photo",
            "a person's arm with a bunch of lipstick swatches on it",
# (snip)
]
img = Image.open(example_image)
inputs = processor(text=candidate_labels, images=img,
 return_tensors="np", padding=True)
ort_inputs = {
    "input_ids": inputs["input_ids"],
    "attention_mask": inputs["attention_mask"],
    "pixel_values": inputs["pixel_values"]
}
ort_outputs = session.run(None, ort_inputs)
```

3. It work but some warinings:
```
2022-10-27 09:23:18.299824270 [W:onnxruntime:, execution_frame.cc:594 AllocateMLValueTensorPreAllocateBuffer] Shape mismatch attempting to re-use buffer. {42,512} != {1,512}. Validate usage of dim_value (values should be > 0) and dim_param (all values with the same string should equate to the same size) in shapes in the model.
2022-10-27 09:23:18.300383069 [W:onnxruntime:, execution_frame.cc:594 AllocateMLValueTensorPreAllocateBuffer] Shape mismatch attempting to re-use buffer. {42,1} != {1,1}. Validate usage of dim_value (values should be > 0) and dim_param (all values with the same string should equate to the same size) in shapes in the model.
```
(`42` is the number of labels)

### Expected behavior

The model work find, but the waring outputs every inference.

**Comment by sgugger** [comment]
cc @lewtun
